pkg/kube/krt/collection.go

```diff
@@ -350,14 +345,10 @@ func (h *manyCollection[I, O]) onPrimaryInputEventLocked(items []Event[I]) {
 	if len(events) == 0 {
 		return
 	}
-	handlers := h.eventHandlers.Get()
-
 	if h.log.DebugEnabled() {
```

Comment:
Why do we have to guard this log? Having it as a `Debugf` isn't enough? Not new in this PR, but just curious

Reply:
The general reason is .WithLabels and sometimes the arguments to it or the log line (`len` here is cheap but imagine it was expensive) have some cost. I don't recall for sure if this initial was more costly or if I had previously decided any `WithLabels` was too costly to run on hot paths